src/scripts/saveToOneNote/oneNoteSaveablePdfSynchronousBatched.ts

```diff
@@ -0,0 +1,65 @@
+import {StringUtils} from "../stringUtils";
+import {ObjectUtils} from "../objectUtils";
+
+import {PdfDocument} from "../contentCapture/pdfDocument";
+
+import {OneNoteSaveable} from "./oneNoteSaveable";
+
+export class OneNoteSaveablePdfSynchronousBatched implements OneNoteSaveable {
+	private page: OneNoteApi.OneNotePage;
+	private pdf: PdfDocument;
+	private pageIndexes: number[];
+	private contentLocale: string;
+	private titleText: string;
+
+	constructor(page: OneNoteApi.OneNotePage, pdf: PdfDocument, pageIndexes: number[], contentLocale: string, titleText: string) {
+		this.page = page;
+		this.pdf = pdf;
+		this.pageIndexes = pageIndexes;
+		this.contentLocale = contentLocale;
+		this.titleText = titleText;
+	}
+
+	/**
+	 * @index starts at 0 and refers to the pages already in the saveable, NOT THE PAGE OF THE PDF ITSELF
+	 */
+	public getPage(index?: number): Promise<OneNoteApi.OneNotePage> {
+		if (ObjectUtils.isNullOrUndefined(index) || index === 0) {
+			// They are asking for the first page
+			return Promise.resolve(this.page);
+		}
+
+		const pageNumber = this.pageIndexes[index - 1];
```

Comment:
I added a lengthy comment that I hope is instructive when I inevitably forget why I did this